### PR TITLE
WE-810 FormationMarker view

### PR DESCRIPTION
**Pull request by janburak**
## Fixes
This pull request fixes WE-810

## Description
Implement formationMarker API routes and view.

## Type of change

* New feature (non-breaking change which adds functionality)

## Impacted Areas in Application

* Frontend, API, WITSML

## Checklist:

*Communication*
* [x] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [ ] New code is covered by passing tests

## Further comments
_If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc..._

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Models/Struct.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+using Witsml.Data;
+
+namespace WitsmlExplorer.Api.Models
+{
+    public class Struct
```
Comment (HakonHesselberg) [inline Src/WitsmlExplorer.Api/Models/Struct.cs]: Struct seems like a name which is too generic. Would StratigraphyStruct be a better choice?

File: Src/WitsmlExplorer.Frontend/components/ContentViews/FormationMarkersListView.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        lithostratigraphic: structToString(formationMarker.lithostratigraphic),
+        chronostratigraphic: structToString(formationMarker.chronostratigraphic),
+        description: formationMarker.description,
+        itemState: formationMarker.commonData.itemState,
+        formationMarker
+      };
+    });
+  };
+
+  const columns: ContentTableColumn[] = [
```
Comment (HakonHesselberg) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/FormationMarkersListView.tsx]: Would it be more efficient to use the useMemo hook here?

```
const columns: ContentTableColumn[] = useMemo(
    () => [
      {property: "name", label: "name", type: ContentType.String},
      {property: "itemState", label: "itemState", type: ContentType.String},
      {property: "mdPrognosed", label: "mdPrognosed", type: ContentType.String},
      {property: "tvdPrognosed", label: "tvdPrognosed", type: ContentType.String},
      {property: "mdTopSample", label: "mdTopSample", type: ContentType.String},
      {property: "tvdTopSample", label: "tvdTopSample", type: ContentType.String},
      {property: "thicknessBed", label: "thicknessBed", type: ContentType.String},
      {property: "thicknessApparent", label: "thicknessApparent", type: ContentType.String},
      {property: "thicknessPerpen", label: "thicknessPerpen", type: ContentType.String},
      {property: "mdLogSample", label: "mdLogSample", type: ContentType.String},
      {property: "tvdLogSample", label: "tvdLogSample", type: ContentType.String},
      {property: "dip", label: "dip", type: ContentType.String},
      {property: "dipDirection", label: "dipDirection", type: ContentType.String},
      {property: "lithostratigraphic", label: "lithostratigraphic", type: ContentType.String},
      {property: "chronostratigraphic", label: "chronostratigraphic", type: ContentType.String},
      {property: "description", label: "description", type: ContentType.String}
    ],
    []
  );
  ```

**Review by HakonHesselberg — DISMISSED** [review]

File: Src/WitsmlExplorer.Frontend/components/ContentViews/FormationMarkersListView.tsx (hunk 2 above)
Comment (janburak) [reply Src/WitsmlExplorer.Frontend/components/ContentViews/FormationMarkersListView.tsx]: Based on [this article](https://blog.logrocket.com/when-not-to-use-usememo-react-hook/#inexpensive-operation), possibly not. Moved `columns` out of the functional component instead, given that the dependency list would be empty.

File: Src/WitsmlExplorer.Api/Models/Struct.cs (hunk 1 above)
Comment (janburak) [reply Src/WitsmlExplorer.Api/Models/Struct.cs]: Good suggestion, implemented.

**Review by HakonHesselberg — APPROVED** [review]

**Pull request merged**